hbase-rsgroup/src/test/java/org/apache/hadoop/hbase/rsgroup/TestRSGroupsKillRS.java

```diff
@@ -131,7 +143,84 @@ public boolean evaluate() throws Exception {
     });
 
     ServerName targetServer1 = getServerName(newServers.iterator().next());
-    Assert.assertEquals(1, admin.getRegions(targetServer1).size());
-    Assert.assertEquals(tableName, admin.getRegions(targetServer1).get(0).getTable());
+    assertEquals(1, admin.getRegions(targetServer1).size());
+    assertEquals(tableName, admin.getRegions(targetServer1).get(0).getTable());
+  }
+
+  @Test
+  public void testKillAllRSInGroup() throws Exception {
+    // create a rsgroup and move one regionserver to it
+    String groupName = "my_group";
+    int groupRSCount = 2;
+    addGroup(groupName, groupRSCount);
+
+    // create a table, and move it to my_group
+    Table t = TEST_UTIL.createMultiRegionTable(tableName, Bytes.toBytes("f"), 5);
+    TEST_UTIL.loadTable(t, Bytes.toBytes("f"));
+    Set<TableName> toAddTables = new HashSet<>();
+    toAddTables.add(tableName);
+    rsGroupAdmin.moveTables(toAddTables, groupName);
+    assertTrue(rsGroupAdmin.getRSGroupInfo(groupName).getTables().contains(tableName));
+    TEST_UTIL.waitTableAvailable(tableName, 30000);
+
+    // check my_group servers and table regions
+    Set<Address> servers = rsGroupAdmin.getRSGroupInfo(groupName).getServers();
+    assertEquals(2, servers.size());
+    LOG.debug("group servers {}", servers);
+    for (RegionInfo tr :
+        master.getAssignmentManager().getRegionStates().getRegionsOfTable(tableName)) {
+      assertTrue(servers.contains(
+          master.getAssignmentManager().getRegionStates().getRegionAssignments()
+              .get(tr).getAddress()));
+    }
+
+    // move all table regions on one group server to another
+    // these codes are aimed to make 'lastHost' in my_group
+    // and check if table regions are online
+    List<ServerName> gsn = new ArrayList<>();
+    for(Address addr : servers){
+      gsn.add(getServerName(addr));
+    }
+    assertEquals(2, gsn.size());
+    for(Map.Entry<RegionInfo, ServerName> entry :
+        master.getAssignmentManager().getRegionStates().getRegionAssignments().entrySet()){
+      if(entry.getKey().getTable().equals(tableName)){
+        LOG.debug("move region {}", entry.getKey().getRegionNameAsString());
```

Comment:
LOG.debug("move region {} from {} to {}", entry.getKey().getRegionNameAsString(), fromServer, toServer);